### PR TITLE
Upgrade to tailwind v2

### PR DESCRIPTION
**Pull request by jaredcwhite**
## Issue

Addresses #1685, #857, and #1684 (also dup #288).

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

The monorepo is upgraded to require Node 14. Both partners and public sites are upgraded to Next 11 (and Webpack 5), plus React is bumped to v17 across the sites and ui-components packages. TailwindCSS is upgraded to v2 along with PostCSS 8. Storybook in ui-components is upgraded to 6.3.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [x] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -117,13 +117,6 @@
   dependencies:
     "@babel/highlight" "^7.10.4"
 
-"@babel/code-frame@7.8.3":
```
Comment (emilyjablonski) [inline yarn.lock:120]: This is a satisfying added/removed diff 😅

**Comment by jaredcwhite** [comment]
Finally got it in a mergeable state with all checks passing! 🙌

**Comment by emilyjablonski** [comment]
Do you have any suggestions for how to test or review this? It LGTM :)

**Comment by seanmalbert** [comment]
@seanmalbert , a note to myself to test if we can easily now use the rtl plugin for tailwind.

**Comment by jaredcwhite** [comment]
@emilyjablonski I'm not sure how much we can verify through tests other than the general infrastructure isn't horribly broken…for me, double-checking the Next/React/Tailwind combo, I just clicked through a bunch of the stuff in Storybook and then various pages on the public/partners sites and did some side-by-side comparisons with the dev branch. Thus far I've resolved the few anomalies I found. Hopefully after a couple other folks giving it a once over, we'll feel reasonablly confident any potential variance is minor and sporadic.

**Review by seanmalbert — APPROVED** [review]
Looks good, @jaredcwhite! I also tested this to make sure tailwind plugins worked with this setup and they do, so I'm going to put out a separate PR for that.

**Comment by seanmalbert** [comment]
I think we can merge this and send to QA at this point so we can have more eyes on any potential variances.

**Comment by jaredcwhite** [comment]
@emilyjablonski Actually I forgot this doesn't include the purge step. Prob don't need to do much A/B comparison until we to #1823

**Pull request merged**